### DISCUSSION TITLE
LazyVim v0.14: blink.cmp, fzf-lua, animations, new which-key preset

### DISCUSSION THREAD

**Discussion by folke** [LazyVim/LazyVim#5166] [Announcements]
Big new release with a lot of changes and improvements!
Two new plugins have been added, and a lot of plugins have been replaced.
With these changes, default **LazyVim** is now just `34` plugins.

### Added Plugins

- [fzf-lua](https://github.com/ibhagwan/fzf-lua) as a replacement for [telescope.nvim](https://github.com/nvim-telescope/telescope.nvim)
  - to use **telescope.nvim** instead, enable the `editor.telescope` extra with `:LazyExtras`
- [blink.cmp](https://github.com/saghen/blink.cmp) as a replacement for [nvim-cmp](https://github.com/hrsh7th/nvim-cmp)
  - to use **nvim-cmp** instead, enable the `coding.nvim-cmp` extra with `:LazyExtras`

### Removed Plugins

- [dressing.nvim](https://github.com/stevearc/dressing.nvim) (replaced with [fzf-lua](https://github.com/ibhagwan/fzf-lua) and [snacks.input](https://github.com/folke/snacks.nvim))
- [telescope-fzf-native.nvim](https://github.com/nvim-telescope/telescope-fzf-native.nvim) (replaced with [fzf-lua](https://github.com/ibhagwan/fzf-lua))
- [telescope.nvim](https://github.com/nvim-telescope/telescope.nvim) (replaced with [fzf-lua](https://github.com/ibhagwan/fzf-lua))
- [indent-blankline.nvim](https://github.com/lukas-reineke/indent-blankline.nvim) (replaced with [snacks.indent](https://github.com/Folke/snacks.nvim))
  - to use **indent-blankline.nvim** instead, enable the `indent-blankline` extra
- [nvim-cmp](https://github.com/hrsh7th/nvim-cmp) (replaced with [blink](https://github.com/Saghen/blink.cmp))
- [nvim-snippets](https://github.com/hrsh7th/vim-vsnip) (replaced with [blink](https://github.com/Saghen/blink.cmp))
- [cmp-buffer](https://github.com/hrsh7th/cmp-buffer) (replaced with [blink](https://github.com/Saghen/blink.cmp))
- [cmp-nvim-lsp](https://github.com/hrsh7th/cmp-nvim-lsp) (replaced with [blink](https://github.com/Saghen/blink.cmp))
- [cmp-path](https://github.com/hrsh7th/cmp-path) (replaced with [blink](https://github.com/Saghen/blink.cmp))

### Changes

- added [`snacks.input`](https://github.com/folke/snacks.nvim/blob/main/docs/input.md)
- added [`snacks.scroll`](https://github.com/folke/snacks.nvim/blob/main/docs/scroll.md)
- added [`snacks.indent`](https://github.com/folke/snacks.nvim/blob/main/docs/indent.md)
- added [`snacks.scope`](https://github.com/folke/snacks.nvim/blob/main/docs/scope.md)
- added [`snacks.dim`](https://github.com/folke/snacks.nvim/blob/main/docs/dim.md)
- added [`snacks.zen`](https://github.com/folke/snacks.nvim/blob/main/docs/zen.md)
- changed default [`which-key`](https://github.com/folke/which-key.nvim) preset to `helix`
  If you prefer the old look and feel:
  ```lua
  return { "folke/which-key.nvim", opts = { preset = "classic" } }
  ```

- drop `LazyVim.ui.fg` in favor of [`Snacks.util.color`](https://github.com/folke/snacks.nvim/blob/main/docs/util.md)

To disable **all animations**, add the following to your `options.lua`:

```lua
vim.g.snacks_animate = false
```

### Keymaps

- `<leader>uz` to toggle [zen mode](https://github.com/folke/snacks.nvim/blob/main/docs/zen.md)
- `<leader>uZ` & `<leader>wm` to toggle [zoom mode](https://github.com/folke/snacks.nvim/blob/main/docs/zen.md)
- `<leader>uD` to toggle [dimming](https://github.com/folke/snacks.nvim/blob/main/docs/dim.md)
- `<leader>ua` to toggle [animations](https://github.com/folke/snacks.nvim/blob/main/docs/animate.md)
- `<leader>uS` to toggle [scroll](https://github.com/folke/snacks.nvim/blob/main/docs/scroll.md)
- `<leader>ug` to toggle [indent guides](https://github.com/folke/snacks.nvim/blob/main/docs/indent.md)
- [`snacks.profiler`](https://github.com/folke/snacks.nvim/blob/main/docs/profiler.md) keymaps under `<leader>dp`

**Comment by bathwaterpizza** [comment]
Enabling animations by default seems a bit odd, seeing as the appeal of Vim is its speed and snappiness. Fantastic update overall though, thanks for all your open source work!

**Reply by dmyyy** [reply]
Agree here - love the new version overall but the animations feel laggy on my relatively minimal lazy config and it took me a little while to find my way here. (at least I know where to look for next time)

**Reply by evanbunnage** [reply]
Some of the animations are neat but any significant visual mode activity is painfully slow.

**Reply by dima-lurie** [reply]
first thing i did was disable the animations... seems counter to the vim experience, though i appreciate the effort. you are amazing ❤️

**Comment by casret** [comment]
Would love to see these https://www.lazyvim.org/configuration/recipes updated to blink.  (I want my emoji source back!)

**Comment by Bobstyle23** [comment]
My custom autocmds for neovim won't work anymore. Do i need to re-create them? Any thoughts?

**Comment by danjessen** [comment]
With `vim.g.snacks_animate = false` I still get the option to enable animation in toggle. Is that intended? I guess the same goes for disabled snacks modules. I've disabled zen mode, but I still the keymaps in toggle and which-key.

**Reply by folke** [reply]
there is nothing to enable/disable for zen mode. same for animate.
Those modules are always available. They don't need a setup or anything like that.

**Comment by StevenXL** [comment]
I am confused as to how I would migrate over to this? Wouldn't closing the starter pack again mean that I lose my customizations? `git clone https://github.com/LazyVim/starter ~/.config/nvim`?

**Reply by danjessen** [reply]
I think you need to elaborate. What specifically are you trying to migrate to ?

**Reply by StevenXL** [reply]
I'm trying to migrate from my current neovim configuration to the new stuff in 0.14. Is that too vague? I thought 0.14 replaced a bunch of stuff that used to be in 0.13. For example, I was still using neotree instead of what is now the default, snacks explorer. What else am I using the "old" version of?

**Reply by dpetka2001** [reply]
If you do the upgrade from Lazy UI then the old defaults will remain the same (neo-tree and fzf-lua). Only if you do a new installation from scratch the defaults will be different. So, you don't have to worry about that.

Make sure you git control your personal configuration in case you might want to revert to a previous state.

**Comment by spundun** [comment]
I did a from scratch install of the main branch(tag v14.15.0) but I don't see blink in my list of plugins. I do already see cmp-buffer/nvim-lsp/path, nvim-cmp and cmp_luasnip loaded.

I don't think the switch from cmp to blink has actually happened as advertised here.

But I'm new to neovim as a whole(even more new to lua) so please let me know if I'm missing something obvious here.
